src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Commands/AddAzureASAccount.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using System.IO;
+using System.Management.Automation;
+using System.Reflection;
+using System.Security;
+using Microsoft.Azure.Commands.AnalysisServices.Dataplane.Properties;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.AnalysisServices.Dataplane
+{
+    /// <summary>
+    /// Cmdlet to log into an Analysis Services environment
+    /// </summary>
+    [Cmdlet("Add", "AzureAnalysisServicesAccount", SupportsShouldProcess=true)]
+    [Alias("Login-AzureAsAccount")]
+    [OutputType(typeof(AsAzureProfile))]
+    public class AddAzureASAccountCommand : AzurePSCmdlet, IModuleAssemblyInitializer
+    {
+        [Parameter(Position = 0, Mandatory = false, HelpMessage = "Name of the Azure Analysis Services environment to which to logon to")]
+        public string RolloutEnvironment { get; set; }
+        
+        [Parameter(Position = 1, Mandatory = false, HelpMessage = "Login credentials to the Azure Analysis Services environment")]
+        public PSCredential Credential { get; set; }
+
+        protected AsAzureEnvironment AsEnvironment;
+           
+        protected override AzureContext DefaultContext
+        {
+            get
+            {
+                // Nothing to do with Azure Resource Managment context
+                return null;
+            }
+        }
+
+        protected override void SaveDataCollectionProfile()
+        {
+            // No data collection for this commandlet 
+        }
+
+        protected override void PromptForDataCollectionProfileIfNotExists()
+        {
+            // No data collection for this commandlet 
+        }
+
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+            if (string.IsNullOrEmpty(RolloutEnvironment))
+            {
+                RolloutEnvironment = AsAzureClientSession.GetDefaultEnvironmentName();
```

Comment:
I strongly encourage you from moving away from a static Session structure - this causes problems when running cmdlets in parallel.  For a future release, you should consider changing this to a Session variable.,